producer/buffer/buffer.go

```diff
@@ -270,6 +294,64 @@ func (b *buffer) cleanupBatchWithLock(
 	return next, removed
 }
 
+func (b *buffer) dropEarliestUntilClose() {
+	ticker := time.NewTicker(b.opts.DropEarliestInterval())
```

Comment:
Do you need the ticker? If you don't aggressively clean up the buffers would you worry about filling the buffer too quickly before it can be cleaned up?

Reply:
I feel it's nicer to have the ticker to batch up some drop work so it's cheaper, we could probably live without it though, when this path is triggered the buffer is already full, a usual config of this ticker would be 1s, when you are buffering for 10 minute, this would mean you spill over 0.16% in that 1 second, which is tiny. Also at the end of the day it's guarded by the `allowedSpilloverRatio` which will enforce a sync drop, so it should be fine here.